services/core/shared/indexdb/mysql.js

```diff
@@ -141,36 +141,46 @@ const getDbInstance = async (tableName, tableConfig, connEndpoint = config.endpo
 		}
 	};
 
-	const queryBuilder = (params, columns) => {
-		const limit = params.limit || 1;
-		const offset = params.offset || 0;
+	const queryBuilder = async (params, columns) => {
+		const limit = Number(params.limit) || 10;
+		const offset = Number(params.offset) || 0;
 
 		delete params.limit;
 		delete params.offset;
 
 		const query = knex.select(columns).table(tableName);
+		const countQuery = knex.count('id as count').table(tableName);
 
 		if (params.propBetween) {
 			const { propBetween } = params;
 			delete params.propBetween;
 			query.whereBetween(propBetween.property, [propBetween.from, propBetween.to]);
-		}
-
-		if (params.orWhere) {
-			const { orWhere } = params;
-			delete params.orWhere;
-			query.orWhere(orWhere);
+			countQuery.whereBetween(propBetween.property, [propBetween.from, propBetween.to]);
 		}
 
 		if (params.sort) {
 			const [sortProp, sortOrder] = params.sort.split(':');
 			delete params.sort;
 			query.orderBy(sortProp, sortOrder);
+			countQuery.orderBy(sortProp, sortOrder);
 		}
 
-		return query.andWhere(params)
-			.limit(limit)
-			.offset(offset);
+		if (params.orWhere) {
+			const { orWhere } = params;
+			delete params.orWhere;
+			query.where(params).orWhere(orWhere);
+			countQuery.where(params).orWhere(orWhere);
+		}
+
+		const [{ count }] = await countQuery.andWhere(params);
```

Comment:
That implementation looks nice, but I'd rather not change the original API.
Can you please move the count logic to the `count(params)` function?
Then we could pass the same params in the app logic.